extensions/amp-jwplayer/1.0/amp-jwplayer.js

```diff
@@ -0,0 +1,117 @@
+import {getDataParamsFromAttributes} from '#core/dom';
+import {dict} from '#core/types/object';
+
+import {isExperimentOn} from '#experiments';
+
+import {userAssert} from '#utils/log';
+
+import {BaseElement} from './base-element';
+
+import {CSS} from '../../../build/amp-jwplayer-1.0.css';
+import {
+  getConsentMetadata,
+  getConsentPolicyInfo,
+  getConsentPolicyState,
+} from '../../../src/consent';
+
+/** @const {string} */
+const TAG = 'amp-jwplayer';
+
+class AmpJwplayer extends BaseElement {
+  /** @override */
+  init() {
+    super.init();
+
+    const consentPolicy = this.getConsentPolicy();
+    if (consentPolicy) {
+      this.getConsentInfo().then(
+        ([policyState, policyInfo, policyMetadata]) => {
+          this.mutateProps(
+            dict({
+              'consentParams': {
+                'policyState': policyState,
+                'policyInfo': policyInfo,
+                'policyMetadata': policyMetadata,
+              },
+            })
+          );
+        }
+      );
+    }
+
+    this.mutateProps(
+      dict({
+        'queryParams': this.mergeQueryParams(
+          getDataParamsFromAttributes(this.element, null, /^playerParam(.+)/),
+          this.element.getAttribute('data-player-querystring')
+        ),
+        'contentSearch': this.getContextualSearch(
+          this.element.getAttribute('data-content-search')
+        ),
+      })
+    );
```

Comment:
I think this can be in the return instead of a call to `mutateProps`:
```suggestion
    return dict({
        'queryParams': this.mergeQueryParams(
          getDataParamsFromAttributes(this.element, null, /^playerParam(.+)/),
          this.element.getAttribute('data-player-querystring')
        ),
        'contentSearch': this.getContextualSearch(
          this.element.getAttribute('data-content-search')
        ),
   });
```

Reply:
updated